### ISSUE TITLE
Building for no_std environment (--no-default-features) fails on nightly.

### ISSUE THREAD

**Issue by jasondavies** [Geal/nom#546]
Apologies if I'm missing something, but when attempting to build for no_std, I get the following errors:

```sh
% cargo build --no-default-features
   Compiling nom v3.2.0 (file:///Users/jason/src/nom)
error[E0432]: unresolved import `collections::boxed`
   --> src/lib.rs:423:25
    |
423 |   pub use collections::{boxed, vec, string};
    |                         ^^^^^ no `boxed` in the root

error[E0412]: cannot find type `Box` in this scope
  --> src/nom.rs:23:39
   |
23 | pub fn tag_cl<'a,'b>(rec:&'a[u8]) ->  Box<Fn(&'b[u8]) -> IResult<&'b[u8], &'b[u8]> + 'a> {
   |                                       ^^^ not found in this scope

error[E0433]: failed to resolve. Use of undeclared type or module `Box`
  --> src/nom.rs:24:3
   |
24 |   Box::new(move |i: &'b[u8]| -> IResult<&'b[u8], &'b[u8]> {
   |   ^^^ Use of undeclared type or module `Box`

error: aborting due to 3 previous errors

error: Could not compile `nom`.

To learn more, run the command again with --verbose.
```

Related #473.

```sh
% rustc --version
rustc 1.21.0-nightly (dd53dd5f9 2017-08-01)
```

**Comment by jasondavies** [comment]
Oh, just discovered #537, which seems to resolve the errors.

**Comment by Geal** [comment]
Hi, the PR has been merged in the "result" branch and will be available for nom 4.0

**Comment by Geal** [comment]
there's been follow-up work on that, and now since 9b12a2d there's an internal module (`lib`) that imports stuff from alloc and std if specified.
The interesting part being that the parts requiring `std` are very reduced now, and you can specify `alloc` to get `many0` and `many1`
